runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -48,22 +60,107 @@
 
 	private GrandExchangePanel panel;
 
+	private MouseListener itemClick;
+
+	@Inject
+	private MouseManager mouseManager;
+
+	@Inject
+	private Client client;
+
 	@Inject
 	private	ClientUI ui;
 
+	@Inject
+	private GrandExchangeConfig config;
+
+	@Provides
+	GrandExchangeConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(GrandExchangeConfig.class);
+	}
+
 	@Override
 	protected void startUp() throws IOException
 	{
 		panel = injector.getInstance(GrandExchangePanel.class);
 		BufferedImage icon = ImageIO.read(getClass().getResourceAsStream("ge_icon.png"));
 		button = new NavigationButton("GE Offers", icon, () -> panel);
 		ui.getPluginToolbar().addNavigation(button);
+
+		itemClick = new MouseListener()
+		{
+			@Override
+			public MouseEvent mouseClicked(MouseEvent e)
```

Comment:
Probably extract this to separate class, this long anonymous implementations should be separate.